packages/cozy-stack-client/src/CozyStackClient.js

```diff
@@ -254,9 +255,22 @@ class CozyStackClient {
     return this.getAuthorizationHeader()
   }
 
+  /**
+   * Change or set the API token
+   *
+   * @param {string|AppToken|AccessToken} token - Stack API token
+   */
   setToken(token) {
-    this.token = token ? new AppToken(token) : null
-    if (token) {
+    if (!token) {
+      this.token = null
+    } else {
+      if (token.toAuthHeader) {
+        // AppToken or AccessToken
+        this.token = token
+      } else {
+        // jwt string
+        this.token = new AppToken(token)
+      }
```

Comment:
Why not the same as `OauthClient` ? :
```jsx 
      setToken(token) {
    if (token) {
      this.token = token instanceof AccessToken ? token : new AccessToken(token)
    } else {
      this.token = null
    }
  }
```

Reply:
Duck typing. I do not see any good reason to limit the token class to AppToken. If someone ever want to send an AccessToken or something else with the same interface, that's fine with me.